zhaquirks/xiaomi/aqara/switch_h1.py

```diff
@@ -0,0 +1,148 @@
+    COMMAND_SINGLE,
+    DEVICE_TYPE,
+    DOUBLE_PRESS,
+    ENDPOINT_ID,
+    ENDPOINTS,
+    INPUT_CLUSTERS,
+    LONG_PRESS,
+    MODELS_INFO,
+    OUTPUT_CLUSTERS,
+    PRESS_TYPE,
+    PROFILE_ID,
+    SHORT_PRESS,
+    VALUE,
+)
+from zhaquirks.xiaomi import (
+    LUMI,
+    BasicCluster,
+    DeviceTemperatureCluster,
+    OnOffCluster,
+    XiaomiMeteringCluster,
+)
+from zhaquirks.xiaomi.aqara.opple_remote import MultistateInputCluster
+from zhaquirks.xiaomi.aqara.opple_switch import OppleSwitchCluster
+
+XIAOMI_COMMAND_SINGLE = "41_single"
+XIAOMI_COMMAND_DOUBLE = "41_double"
+XIAOMI_COMMAND_HOLD = "1_hold"
+
+
+class AqaraH1SingleRockerSwitch(CustomDevice):
+    """Aqara H1 Single Rocker Switch (with neutral)."""
+
+    signature = {
+        MODELS_INFO: [(LUMI, "lumi.switch.n1aeu1")],
+        ENDPOINTS: {
+            #  input_clusters=[0, 2, 3, 4, 5, 6, 18, 64704], output_clusters=[10, 25]
+            1: {
+                PROFILE_ID: zha.PROFILE_ID,
+                DEVICE_TYPE: zha.DeviceType.ON_OFF_LIGHT,
+                INPUT_CLUSTERS: [
+                    Basic.cluster_id,  # 0
+                    DeviceTemperatureCluster.cluster_id,  # 2
+                    Identify.cluster_id,  # 3
+                    Groups.cluster_id,  # 4
+                    Scenes.cluster_id,  # 5
+                    OnOff.cluster_id,  # 6
+                    Alarms.cluster_id,  # 9
+                    XiaomiMeteringCluster.cluster_id,  # 0x0702
+                    0x0B04,
+                ],
+                OUTPUT_CLUSTERS: [
+                    Time.cluster_id,  # 0x000a
+                    Ota.cluster_id,  # 0x0019
+                ],
+            },
+            242: {
+                PROFILE_ID: 41440,
+                DEVICE_TYPE: 0x0061,
+                INPUT_CLUSTERS: [],
+                OUTPUT_CLUSTERS: [
+                    GreenPowerProxy.cluster_id,  # 0x0021
+                ],
+            },
+        },
+    }
+
+    replacement = {
+        ENDPOINTS: {
+            1: {
+                PROFILE_ID: zha.PROFILE_ID,
+                DEVICE_TYPE: zha.DeviceType.ON_OFF_SWITCH,
+                INPUT_CLUSTERS: [
+                    BasicCluster,  # 0
+                    DeviceTemperatureCluster.cluster_id,  # 2
+                    Identify.cluster_id,  # 3
+                    Groups.cluster_id,  # 4
+                    Scenes.cluster_id,  # 5
+                    OnOffCluster,  # 6
+                    Alarms.cluster_id,  # 9
+                    MultistateInputCluster,  # 18
+                    XiaomiMeteringCluster.cluster_id,  # 0x0702
+                    OppleSwitchCluster,  # 0xFCC0 / 64704
+                    0x0B04,
+                ],
+                OUTPUT_CLUSTERS: [
+                    Time.cluster_id,
+                    Ota.cluster_id,
+                ],
+            },
+            41: {
+                PROFILE_ID: zha.PROFILE_ID,
+                DEVICE_TYPE: zha.DeviceType.ON_OFF_SWITCH,
```

Comment:
Hopefully one last thought:

You're changing the device type here.
Simple switch functionality already works without the quirk, right?
So this could be a breaking change for existing users.

The type can also be changed in the `configuration.yaml` without using a quirk (IEEE and endpoint needed).

At least for Tuya (light) switches, we've tried to avoid changing the device type using quirks.

Any opinions on changing the device type for this device?
Honestly, the change shouldn't be too breaking, but we should still consider if we want to make it.

Reply:
I understand your concern. However, I'm basing this quirk on the [double rocker version](https://zigbee.blakadder.com/Aqara_WS-EUK04.html) of the same type of switch which is already being actively used and, judging from all the comments from the community, people are keen on having the same functionality for the [single rocker switch](https://zigbee.blakadder.com/Aqara_WS-EUK03.html) as well which I happen to own so I figured why don't I give something back to the community.

My personal preference also matches the existing implementation for the double rocker switch which is to expose them as switches instead of lights. For reference: https://github.com/zigpy/zha-device-handlers/blob/dev/zhaquirks/xiaomi/aqara/opple_switch.py#L124

Reply:
<img width="991" alt="Screenshot 2023-05-22 at 21 22 01" src="https://github.com/zigpy/zha-device-handlers/assets/4711683/96567efd-c081-4131-b339-2dbbda80d9a0">



Time for these bad boys to get some much deserved support 🙂

Reply:
Since it's consistent with the other switches, it's probably fine. 👍